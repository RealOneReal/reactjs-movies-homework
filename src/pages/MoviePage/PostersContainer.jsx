import { Grid } from "@mui/material";
import { images } from "../../mock-data/mockMovieDetail";

const PostersContainer = () => {
  const posters = images.posters?.filter((_img, index) => index < 8);

  return (
    <Grid container gap={1}>
      {posters.map((img) => (
        <Grid
          item
          sx={{
            backgroundImage: `url(${img.file_path})`,
            backgroundPosition: "center center",
            backgroundSize: "contain",
            backgroundRepeat: "no-repeat",
            width: "210px",
            height: "120px",
          }}
        />
      ))}
    </Grid>
  );
};

export default PostersContainer;
