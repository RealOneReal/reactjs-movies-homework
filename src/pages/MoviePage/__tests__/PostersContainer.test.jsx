import { render } from "@testing-library/react";
import PostersContainer from "../PostersContainer";

describe("PostersContainer", () => {
  it("should be rendered", () => {
    const { container } = render(<PostersContainer />);
    expect(container).toMatchSnapshot();
  });
});
