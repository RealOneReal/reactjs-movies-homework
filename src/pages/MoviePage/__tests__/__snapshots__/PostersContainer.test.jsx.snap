// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostersContainer should be rendered 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-1dfafci-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-1ct45ut-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-5fjczf-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1ct45ut-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1f730dw-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1ct45ut-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1f730dw-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1ct45ut-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item css-1f730dw-MuiGrid-root"
    />
  </div>
</div>
`;
